source/mir/primitives.d

```diff
@@ -367,3 +367,28 @@ unittest
     auto a = [1, 2];
     assert(a is a.save);
 }
+
+/++
+Returns: `true` if `R` has a `elementCount` member that returns an
+integral type implicitly convertible to `size_t`.
+
+`R` does not have to be a range.
++/
+enum bool hasElementCount(R) = is(typeof(
```

Comment:
Do we really need this? Why not to use `hasShape` instead?

Reply:
The mean function currently does not work for N-dimesnional slices when N > 1. In numir ([here](https://github.com/libmir/numir/blob/d47d6b5213f6a322f4bffc17d433148fec268c02/source/numir/stats.d#L193)) I used elementCount successfully to get this to work. My WIP for enhancing the mean function is [here](https://github.com/jmh530/mir-algorithm/commit/8f6c2c372d2427b8f7769d066a4b11bee311f9ae). Happy to do something else if it works better.

Reply:
`hasShape` should work instead